crates/cdk/src/mint/melt.rs

```diff
@@ -93,6 +123,14 @@ impl Mint {
             Error::UnitUnsupported
         })?;
 
+        // check amountless (amount in meltbolt11request) is equal to payment_quote.amount
```

Comment:
Why do we need to check this, shouldn't we be setting the amount?

Reply:
do you mean mint return an amount, 

if this is the case, mint does not know the amount of ecash a wallet making the melt qoute request has.

Reply:
if mint is setting the amount coming from the option field in `MeltQuoteBolt11Request` then , we need to check to ensure only mint that support amountless payment can handle this.